### PR TITLE
py-fenics-ffcx: update to v0.7

### PR DESCRIPTION
**Pull request by garth-wells**
Note that numpy>= 1.21 in the Spack recipe. This is correct because FFCx uses Numpy typing, which was introduced in Numpy 1.21. The Numpy version is not properly constrained in FFCx (fixed in FFCx dev branch).

Also adds tests.

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the new version sha256.

**Comment by tldahlgren** [comment]
I was also able to successfully install the package.

**Review by chrisrichardson — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-fenics-ffcx/package.py [hunk 1]
```diff
@@ -15,24 +15,39 @@ class PyFenicsFfcx(PythonPackage):
...
     depends_on("py-fenics-ufl@2022.2.0", type=("build", "run"), when="@0.5.0:0.5")
     depends_on("py-fenics-ufl@2022.1.0", type=("build", "run"), when="@0.4.2")
 
     depends_on("py-fenics-basix@main", type=("build", "run"), when="@main")
+    depends_on("py-fenics-basix@0.7", type=("build", "run"), when="@0.7")
     depends_on("py-fenics-basix@0.6.0:0.6", type=("build", "run"), when="@0.6.0:0.6")
     depends_on("py-fenics-basix@0.5.1:0.5", type=("build", "run"), when="@0.5.0:0.5")
     depends_on("py-fenics-basix@0.4.2", type=("build", "run"), when="@0.4.2")
+
+    depends_on("py-pytest@0.6:", type="test")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fenics-ffcx/package.py]:
```suggestion
    depends_on("py-pytest@6:", type="test")
```

File: var/spack/repos/builtin/packages/py-fenics-ffcx/package.py [hunk 2]
```diff
@@ -15,24 +15,39 @@ class PyFenicsFfcx(PythonPackage):
...
 
-    depends_on("python@3.7:", type=("build", "run"))
-    depends_on("py-setuptools@58:", type=("build", "run"))
+    depends_on("python@3.8:", when="@0.7:", type=("build", "run"))
+    depends_on("py-setuptools@62:", when="@0.7:", type="build")
+    depends_on("py-setuptools@58:", when="@0.4.2:0.6", type="build")
 
+    # CFFI is required at runtime for JIT support
     depends_on("py-cffi", type=("build", "run"))
     depends_on("py-numpy@1.21:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-fenics-ffcx/package.py:33]: Can you add a comment about type hints being the reason for the version req so I don't forget in the next PR?

Comment (garth-wells) [reply var/spack/repos/builtin/packages/py-fenics-ffcx/package.py:33]: Done.

**Comment by garth-wells** [comment]
@adamjstewart is this PR now good to go?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**